lib/merit/order.rb

```diff
@@ -118,16 +118,7 @@ def profit_info
     # Public: Returns an Array containing a 'table' with all the producers
     # vertically, and horizontally the power per point in time.
     def load_curves
-      columns = []
-      participants.producers.each do |producer|
-        columns << [producer.key,
-                    producer.class,
-                    producer.output_capacity_per_unit,
-                    producer.number_of_units,
-                    producer.load_curve.to_a
-        ].flatten
-      end
-      columns.transpose
+      participants.map(&:to_csv_row).transpose
```

Comment:
I don't think converting to a CSV representation is really the the responsibility of the producer or user (which are concerned with the calculation of load). It feels like it's breaking the single responsibility principle to do this. How about a simple ParticipantSummary module instead?

```ruby
module ParticipantSummary
  module_function

  def summarize(order)
    order.participants.map do |participant|
      case participant
        when User then summarize_user(participant)
        else           summarize_participant(participant)
      end
    end
  end

  def summarize_user(user)
    # ...
  end

  def summarize_participant(participant)
    # ...
  end
end
```